Src/GoogleApis.Auth/OAuth2/ServiceCredential.cs

```diff
@@ -197,6 +187,29 @@ public void Initialize(ConfigurableHttpClient httpClient)
 
         #endregion
 
+
+        #region ITokenAccess implementation
+
+        /// <summary>
+        /// Gets an access token to authorize a request. If the existing token has expired, try to refresh it first.
+        /// <seealso cref="ITokenAccess.GetAccessTokenForRequestAsync"/>
+        /// </summary>
+        public virtual async Task<string> GetAccessTokenForRequestAsync(string authUri = null, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (Token == null || Token.IsExpired(Clock))
+            {
+                Logger.Debug("Token has expired, trying to get a new one.");
+                if (!await RequestAccessTokenAsync(cancellationToken).ConfigureAwait(false))
+                {
+                    throw new InvalidOperationException("The access token has expired but we can't refresh it");
+                }
+                Logger.Info("New access token was received successfully");
+            }
+            return Token.AccessToken;

```

Comment:
This is actually preexisting code, I just refactored the call sites a bit. I think it's fine without the second null check because once RequestAccessTokenAsync  succeeds, it will always set the Token. If it fails, this method throws exception before it reaches line 208.